src/pd/async/util.rs

```diff
@@ -74,34 +82,52 @@ impl LeaderClient {
     }
 
     // Re-establish connection with PD leader in synchronized fashion.
-    pub fn reconnect(&self) {
-        let start = Instant::now();
-        let ret = try_connect_leader(&self.inner.rl().members);
+    pub fn reconnect(&self) -> PdFuture<()> {
```

Comment:
It mean a new connection __should__ be established. Actually since `try_connect_leader` is synchronized, so at the time this function return, a connection has been established.